src/main/java/frc/robot/commands/DefaultDriveCommand.java

```diff
@@ -29,11 +29,11 @@ public void execute() {
     double rot = -Input.getRot();
 
     if (Input.resetGyro()) {
-      mSwerve.resetGyro();
+      PoseEstimation.resetGyro();
```

Comment:
So based on this calling pattern, PostEstimation operates on global state? Basically, global variables?

Reply:
Yup! Pose Estimation is just the object that accumulates the pose reading from the swerve and vision, which we decided to make global to easily reset and read the pose.

Reply:
Global state is almost always the wrong answer and should generally be avoided as much as possible.

When you have global state, it's hard to test. It's hard to create a duplicate of the state if you need to (you can easily have 2 objects of some type, but you can't do that with global state). Global state may also lead to mutual exclusion issues.

Is it too late to move away from this pattern and treat those as objects instead?

Reply:
Hmm, I see what you mean, rather than making the whole class non-static, would changing the resetPose and resetGyro methods from being static to just instance based fix this issue since those 2 methods are the only way to change the state of the class?

Reply:
Yes, I think that would help.

Reply:
Yesterday, we ended up moving PoseEstimation to swerve(which isn't a big addition) and made the object itself and certain methods that access/reset the pose static. Certain commands, such as pathplanner, require the methods that handle the pose to be static, so we couldn't make it instance-based. Is that okay?